pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -568,6 +568,18 @@ func fitsResourceQuota(fName kueue.ResourceFlavorReference, rName corev1.Resourc
 	if cq.Cohort != nil {
 		cohortUsed = cq.Cohort.Usage[fName][rName]
 		cohortAvailable = cq.Cohort.RequestableResources[fName][rName]
+		// if LendingLimit enabled, we need to recalculate cohort.RequestableResources
+		if features.Enabled(features.LendingLimit) {
+			// to borrow from the next flavor, add " && cq.FlavorFungibility.WhenCanBorrow == kueue.TryNextFlavor" here
+			if rQuota.LendingLimit != nil {
+				cohortAvailable += rQuota.Nominal - *rQuota.LendingLimit
+			}
+
+			if used+val > cohortAvailable {
+				status.append(fmt.Sprintf("lending limit for %s in flavor %s exceeded", rName, fName))
```

Comment:
How can we know we're exceeding the lendingLimit, what if other clusterQueues didn't set this? Let's just pass through the return.